packages/api/src/controllers/saleController.js

```diff
@@ -53,8 +54,9 @@ const SaleController = {
         saleData.sale_date = sale_date;
       }
 
-      // TODO: LF-3595 - properly determine if value and note need to be updated
-      if (!req.body.crop_variety_sale?.length) {
+      const revenueType = await RevenueTypeModel.query().findById(revenue_type_id);
```

Comment:
So if we don't add this to the try/catch with trx then we are opening 2 trx in this method I think.

Reply:
Sequentially though, right? So it wouldn't have any effect on filling the transaction pool -- this one is closed before the try/catch one is opened.